packages/fetchai/connections/soef/connection.py

```diff
@@ -560,10 +602,13 @@ async def _ping_periodic(self, period: float = 30 * 60) -> None:
         :return: None
         """
         with suppress(asyncio.CancelledError):
-            while True:
+            while self.unique_page_address:
```

Comment:
Initially, this will likely be None, so the ping task might never start.

Reply:
it starts only after register and keep registration alive.
no reason to ping with no registration completed